### PR TITLE
Pulsed mb

### PR DESCRIPTION
**Pull request by kaelyndunnell**
Added pulses fluxes and temperatures. Integrated with DINA data. Added recombination BC on surface.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -6,5 +6,5 @@ dependencies:
   - fenics-dolfinx=0.8.0
```
Comment (RemDelaporteMathurin) [inline environment.yml:6]: Can You upgrade this to 0.9?

File: environment.yml [hunk 2]
```diff
@@ -6,5 +6,5 @@ dependencies:
...
   - pip 
   - pip:
-    - git+https://github.com/festim-dev/FESTIM@77c97b30b7b9db90cc35754c8228b3f8f52a7f6d
+    - git+https://github.com/remdelaportemathurin/FESTIM@surface-reactions
```
Comment (RemDelaporteMathurin) [inline environment.yml:9]: You may want to pin a specific commit

File: simple_mb.py [hunk 3]
```diff
@@ -1,16 +1,72 @@
...
+        self.flux_fenics = Constant(mesh, float(self.flux(t)))
+        x = ufl.SpatialCoordinate(mesh)
+        self.distribution_fenics = self.distribution(x)
+
+        self.value_fenics = self.flux_fenics * self.distribution_fenics
+
+    def update(self, t: float):
+        self.flux_fenics.value = self.flux(t)
+
+class FluxFromSurfaceReaction(F.SurfaceFlux):
```
Comment (RemDelaporteMathurin) [inline simple_mb.py:35]: If you don't use this class please remove it

File: simple_mb.py [hunk 4]
```diff
@@ -1,16 +1,72 @@
...
+
+# TODO: ADJUST TO HANDLE ANY STRAIGHT W 6MM SIMU
+mb = 50
+
+############# Input Flux, Heat Data #############
+with open('scenario.txt', 'r') as file:
+    lines = []
+    for line in file:
+        lines.append(line.split())
+lines = lines[1:]
```
Comment (RemDelaporteMathurin) [inline simple_mb.py:57]: You should use numpy genfromtext function instead of this

File: simple_mb.py [hunk 5]
```diff
@@ -20,17 +76,17 @@
...
+# W material parameters 
+# TODO integrate with htm 
 w_density = 6.3382e28  # atoms/m3
-tungsten_diff = htm.diffusivities.filter(material=htm.TUNGSTEN).mean()
+# tungsten_diff = htm.diffusivities.filter(material=htm.TUNGSTEN).mean()
 tungsten = F.Material(
-    D_0=tungsten_diff.pre_exp.magnitude,
-    E_D=tungsten_diff.act_energy.magnitude,
+    D_0=4.1e-7, # Hodille 2015 paper pg 426
+    E_D=0.39,
```
Comment (RemDelaporteMathurin) [inline simple_mb.py:85]: This is not @ehodille but from Frauenfeldeer, we have this property in HTM

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]

File: environment.yml
Comment (RemDelaporteMathurin) [inline environment.yml:1]: You need to add matplotlib as a conda dependency. This is what is making the Ci fail. Here, matplotlib is installed alongside HTM as a pip package. This conflicts with dolfinx which is installed with Conda.

(hello @shimwell)

**Comment by kaelyndunnell** [comment]
Solves issue #7 , issue #8 , issue  #9 , and issue #10

**Pull request merged**